homeassistant/components/sensor/time_date.py

```diff
@@ -56,7 +59,7 @@ def __init__(self, option_type):
         self._name = OPTION_TYPES[option_type]
         self.type = option_type
         self._state = None
-        self.update()
+        self.async_update()
```

Comment:
You can only call async_update from a coro with yield from or create_task. You should call `yield from async_update` after create on `async_setup` and remove here. It is not possible to have a coroutine as init and you can call async coro stuff from init. That is the point of helper init function in group component